flocker/provision/_digitalocean.py

```diff
@@ -0,0 +1,223 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+DigitalOcean provisioner.
+"""
+import time
+from functools import partial
+
+import pyocean
+
+from ._libcloud import LibcloudProvisioner
+from ._install import (
+    provision, run,
+    task_install_kernel,
+)
+
+
+def retry_if_pending(callable, *args, **kwargs):
+    """
+    DigitalOceanV2 API only allows one change at a time and returns HTTP code
+    402 if another change is already pending.
+
+    So this function repeats the API call if that error code is received and
+    returns the result if the call eventually succeeds.
+
+    The raw DO API returns ``event``s whose status can be queried, and that
+    would be a better way to block before issuing the next API call, but
+    pyocean doesn't consistently return the event info. E.g. droplet.create
+    returns a ``droplet`` instance instead whose status is difficult to check.
+
+    See https://digitalocean.uservoice.com/forums/136585-digitalocean/suggestions/4842992-allow-api-calls-to-queue-rather-than-just-rejectin # noqa
+    """
+    while True:
+        try:
+            result = callable(*args, **kwargs)
+        except pyocean.exceptions.ClientError as e:
+            if e.message == 'Droplet already has a pending event.':
+                time.sleep(1)
+                continue
+            raise
+        else:
+            return result
+
+
+def set_latest_droplet_kernel(droplet, kernel_prefix='Fedora 20 x64'):
```

Comment:
Add a note here about the known-working kernel. I imagine that will be useful for debugging if a future kernel causes problems.